lib/src/embedded/message-transformer.ts

```diff
@@ -83,25 +81,11 @@ export class MessageTransformer {
     private readonly outboundProtobufs$: Observable<Buffer>,
     private readonly writeInboundProtobuf: (buffer: Buffer) => void
   ) {
-    this.outboundProtobufs$.subscribe(
-      buffer => {
-        try {
-          this.outboundMessagesInternal$.next(decode(buffer));
-        } catch (error) {
-          this.error$.next(error);
-        }
-      },
-      error => this.error$.next(error),
-      () => {
-        this.outboundMessagesInternal$.complete();
-        this.error$.complete();
-      }
+    this.outboundProtobufs$.pipe(map(buffer => decode(buffer))).subscribe(
+      buffer => this.outboundMessagesInternal$.next(buffer),
+      error => this.outboundMessagesInternal$.error(error),
+      () => this.outboundMessagesInternal$.complete()
```

Comment:
```suggestion
      this.outboundMessagesInternal$
```

`Observable.subscribe()` is intentionally able to take an `Observer` object and call its methods.